python/ray/data/_internal/table_block.py

```diff
@@ -46,6 +48,20 @@ def add(self, item: Union[dict, TableRow, np.ndarray]) -> None:
                 "Returned elements of an TableBlock must be of type `dict`, "
                 "got {} (type {}).".format(item, type(item))
             )
+
+        item_columns_names = sorted(item.keys())
+        if self._sorted_columns_names:
```

Comment:
This should probably be an `is not None` check rather than a truthy check, in case all previously added rows were empty.

Reply:
@clarkzinzow - good catch! updated.